cranelift-codegen/meta/src/shared/instructions.rs

```diff
@@ -2630,6 +2634,44 @@ pub fn define(
         .operands_out(vec![a]),
     );
 
+    let x = &operand("x", Any);
+    let a = &operand_doc("a", AnyTo, "Bits of `x` reinterpreted");
+
+    ig.push(
+        Inst::new(
+            "raw_bitcast",
```

Comment:
Maybe name it `simd_bitcast`?

Reply:
In talking to @sunfishcode I got the impression that this is not a SIMD-only instruction; presumably it could be used on scalar types to convert one SSA type to another without changing the bits.

Reply:
The difference between this and `bitcast` is the ability to change lane amount right?

Reply:
I don't think so: I originally implemented all of this behavior under `bitcast` (and I was able to change the number of lanes just fine) but @sunfishcode wanted me to create a new instruction because `bitcast` has the idea of changing the bits and what I was doing was more a true cast--no bit changes.

Reply:
Right. It's not just about changing the lane amount; it's also about the fact that boolean types in cranelift are opaque, and we don't want to expose their bit patterns. There are operations we want to do on such types, such as shuffling the lanes, which don't ultimately expose their bit patterns, but it's difficult to express that in the type system. So I have a slight preference for `raw_bitcast` here.